download-utils/src/lib.rs

```diff
@@ -53,28 +59,51 @@ pub fn download_file(url: &str, destination_file: &Path) -> Result<(), String> {
             .and_then(|content_length| content_length.parse().ok())
             .unwrap_or(0)
     };
-    progress_bar.set_length(download_size);
-    progress_bar.set_style(
-        ProgressStyle::default_bar()
-            .template(&format!(
-                "{}{}Downloading {} {}",
-                "{spinner:.green} ",
-                TRUCK,
-                url,
-                "[{bar:40.cyan/blue}] {bytes}/{total_bytes} ({eta})"
-            ))
-            .progress_chars("=> "),
-    );
+
+    if use_progress_bar {
+        progress_bar.set_length(download_size);
+        progress_bar.set_style(
+            ProgressStyle::default_bar()
+                .template(&format!(
+                    "{}{}Downloading {} {}",
+                    "{spinner:.green} ",
+                    TRUCK,
+                    url,
+                    "[{bar:40.cyan/blue}] {bytes}/{total_bytes} ({eta})"
+                ))
+                .progress_chars("=> "),
+        );
+    } else {
+        info!("Downloading {} bytes from {}", download_size, url);
+    }
 
     struct DownloadProgress<R> {
         progress_bar: ProgressBar,
         response: R,
+        last_print: Instant,
+        current_bytes: usize,
+        download_size: f32,
```

Comment:
Probably want `f64`.

Or is this how we're going to prioritize keeping snapshots small? :slightly_smiling_face:

Reply:
f32 can represent large sizes, it just means that those will be rounded in the least significant digits and obviously f32 is more imprecise the larger the number, but that preciseness doesn't matter as much. I don't think it really matters for the user if they see 21.2% versus 21.3% downloaded as their status, or distinguishing a speed of 100.1mbps versus 100.2mbps.

Reply:
Largest normal is this:
`0 11111110 111111111111111111111112 = 7f7f ffff16 = 2127 × (2 − 2−23) ≈ 3.4028234664 × 1038`

If snapshot sizes are that large, we are in trouble.

Reply:
Ah, it's only used in math.  I'd misread and thought we were printing it directly